contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -2773,7 +2773,13 @@ TdsPrintTup(TupleTableSlot *slot, DestReceiver *self)
 		 * NBCROW (0xD2). Count the number of nullable columns and build the
 		 * null bitmap just in case while we are at it.
 		 */
-		nullMapSize = (natts + 7) / 8;
+
+		if (sendRowStat)
+			/* Extra bit for the ROWSTAT column */
+			nullMapSize = (natts >> 3) + 1;
+		else
+			nullMapSize = ((natts - 1) >> 3) + 1;
```

Comment:
this change does not match with problem description. Also, please add a test case.

Reply:
Updated the expression to avoid negative values in case `natts` is 0 as discussed offline. Updated the description regarding the test cases